python/friesian/example/recall/README.md

```diff
@@ -1,6 +1,100 @@
+conda create -n bigdl python=3.7  # "bigdl" is the conda environment name, you can use any name you like.
+conda activate bigdl
+conda install -c pytorch faiss-cpu 
+pip install --pre --upgrade bigdl-friesian
+```
+
+## Generate data
+You can generate some test data for testing, example command:
+```bash
+python generate_test_data.py \
+    --row_nums 200000 \
+    --vec_dim 256 \
+    --use_spark \
+    --verbose \
+    --index_type FlatL2 \
+    --emb_path /path/to/save/vector/embeddings \
+    --dict_path /path/to/save/item_dict \
+    --index_save_path /path/to/save/faiss/index/data \
+    --parquet_path /path/to/save/vector/embeddings/in/parquet/
 ```
-conda install -c pytorch faiss-cpu
+
+## Retrieving items
+* Spark local, example command:
+```bash
+python search.py \
+    --num_threads 8 \
+    --num_repartition 4 \
+    --cluster_mode local \
+    --top_k 100 \
+    --batch_size 50000 \
+    --dict_path /path/to/the/folder/of/item_dict \
+    --faiss_index_path /path/to/the/folder/of/faiss/index/data \
+    --parquet_path /path/to/the/folder/of/vector/embeddings/in/parquet \
+    --parquet_output_path /path/to/the/folder/to/save/retrieval/results 
 ```
+
+* Spark spark-submit, example command:
+```bash
+python search.py \
+    --num_threads 8 \
+    --num_repartition 4 \
+    --cluster_mode spark-submit \
+    --top_k 100 \
+    --batch_size 50000 \
+    --dict_path /path/to/the/folder/of/item_dict \
+    --faiss_index_path /path/to/the/folder/of/faiss/index/data \
+    --parquet_path /path/to/the/folder/of/vector/embeddings/in/parquet \
+    --parquet_output_path /path/to/the/folder/to/save/retrieval/results 
+```
+
+* Spark yarn client mode, example command:
+```bash
+python search.py \
+    --num_threads 8 \
+    --num_repartition 12 \
+    --cluster_mode yarn \
+    --num_nodes 4 \
+    --executor_cores 8 \
+    --executor_memory 50g \
+    --top_k 100 \
+    --batch_size 50000 \
+    --dict_path /path/to/the/folder/of/item_dict \
+    --faiss_index_path /path/to/the/folder/of/faiss/index/data \
+    --parquet_path /path/to/the/folder/of/vector/embeddings/in/parquet \
+    --parquet_output_path /path/to/the/folder/to/save/retrieval/results 
+```
+
+__Options for generate_test_data:__
+* `row_nums`: The number of vectors to be generated. Default to be 200000.
+* `vec_dim`: The dimension of vector. Default to be 256.
+* `use_spark`: Use spark to generate vector embeddings, saving to parquet file. Default to be False.
+* `verbose`: Print more detail information. Default to be False.
+* `index_type`: The faiss index_type: FlatL2 or IVFFlatL2. Default to be FlatL2.
+* `emb_path`: The path to save vector embeddings. Default to be ./emb_vecs.pkl.
+* `dict_path`: The path to save item_dict. Default to be ./item_dict.pkl.
+* `index_save_path`: The path to save faiss index data. Default to be ./index_FlatL2.pkl.
+* `parquet_path`: The path to save vector embeddings with spark, only work when use_spark is True. Default to be ./data.parquet/.
```

Comment:
Put this into generate data

Reply:
Also need to briefly describe these paths would be the input of search.py?

Reply:
ok~